.cirrus.yml

```diff
@@ -39,7 +39,7 @@ main_template: &MAIN_TEMPLATE
   ci_script:
     - ./ci/test_run_all.sh
 
-global_task_template: &GLOBAL_TASK_TEMPLATE
+container_depends_template: &CONTAINER_DEPENDS_TEMPLATE
```

Comment:
Any reason to add a new template when the existing can be used? (Untested diff):


```diff
diff --git a/.cirrus.yml b/.cirrus.yml
index 232b24117a..9367d50f6b 100644
--- a/.cirrus.yml
+++ b/.cirrus.yml
@@ -39,7 +39,7 @@ main_template: &MAIN_TEMPLATE
   ci_script:
     - ./ci/test_run_all.sh
 
-container_depends_template: &CONTAINER_DEPENDS_TEMPLATE
+global_task_template: &GLOBAL_TASK_TEMPLATE
   << : *BASE_TEMPLATE
   container:
     # https://cirrus-ci.org/faq/#are-there-any-limits
@@ -50,9 +50,6 @@ container_depends_template: &CONTAINER_DEPENDS_TEMPLATE
   depends_built_cache:
     folder: "depends/built"
     fingerprint_script: echo $CIRRUS_TASK_NAME $(git rev-list -1 HEAD ./depends)
-
-global_task_template: &GLOBAL_TASK_TEMPLATE
-  << : *CONTAINER_DEPENDS_TEMPLATE
   << : *MAIN_TEMPLATE
 
 compute_credits_template: &CREDITS_TEMPLATE
@@ -303,13 +300,12 @@ task:
 
 task:
   name: 'macOS 10.15 [gui, no tests] [focal]'
-  << : *CONTAINER_DEPENDS_TEMPLATE
-  container:
-    image: ubuntu:focal
   macos_sdk_cache:
     folder: "depends/SDKs/$MACOS_SDK"
     fingerprint_key: "$MACOS_SDK"
-  << : *MAIN_TEMPLATE
+  << : *GLOBAL_TASK_TEMPLATE
+  container:
+    image: ubuntu:focal
   env:
     MACOS_SDK: "Xcode-12.2-12B45b-extracted-SDK-with-libcxx-headers"
     << : *CIRRUS_EPHEMERAL_WORKER_TEMPLATE_ENV
@@ -334,16 +330,15 @@ task:
 
 task:
   name: 'ARM64 Android APK [jammy]'
-  << : *CONTAINER_DEPENDS_TEMPLATE
-  container:
-    image: ubuntu:jammy
   android_sdk_cache:
     folder: "depends/SDKs/android"
     fingerprint_key: "ANDROID_API_LEVEL=28 ANDROID_BUILD_TOOLS_VERSION=28.0.3 ANDROID_NDK_VERSION=23.2.8568313"
   depends_sources_cache:
     folder: "depends/sources"
     fingerprint_script: git rev-list -1 HEAD ./depends
-  << : *MAIN_TEMPLATE
+  << : *GLOBAL_TASK_TEMPLATE
+  container:
+    image: ubuntu:jammy
   env:
     << : *CIRRUS_EPHEMERAL_WORKER_TEMPLATE_ENV
     FILE_ENV: "./ci/test/00_setup_env_android.sh"

Reply:
> Untested diff

It looks correct and elegant.